llvm/lib/Transforms/InstCombine/InstCombineInternal.h

```diff
@@ -433,7 +433,7 @@ class LLVM_LIBRARY_VISIBILITY InstCombinerImpl final
   Value *foldAndOrOfICmpsOfAndWithPow2(ICmpInst *LHS, ICmpInst *RHS,
                                        Instruction *CxtI, bool IsAnd,
                                        bool IsLogical = false);
-  Value *matchSelectFromAndOr(Value *A, Value *B, Value *C, Value *D,
+  Value *matchSelectFromAndOr(Value *A, Value *C, Value *B, Value *D,
```

Comment:
What the <expletive> is happening here?

...I understand that this commit adjusts the declaration to match the names appearing in the definition, but why would the definition use the names "A", "C", "B" and "D" in this order?

This might be a bug in the definition of this function -- and even if it's not, I think it would be worth to preserve the natural ordering of these letters and adjust the definition of the function accordingly. (Of course, using descriptive variable names instead of meaningless single letters would be even better...)

Reply:
The pre-commit version of the function declaration listed the parameters in the order A, C, B, D to indicate the expected grouping of arguments, i.e. (A & C) | (B & D). Changing the order to A, B, C, D would have required reworking the function, which seemed outside the scope of this issue - so I opted to reflect the A, C, B, D order in the function declaration instead. However, as requested, I have modified the parameter names, listed them in logically ascending order, and reworked the function to match.